opendevin/runtime/plugins/agent_skills/agentskills.py

```diff
@@ -68,6 +69,53 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _is_valid_filename(file_name) -> bool:
+    if not file_name or not isinstance(file_name, str) or not file_name.strip():
+        return False
+    # Define invalid characters for the current operating system
+    invalid_chars = '<>:"/\\|?*'
+    if os.name == 'nt':  # Windows
+        invalid_chars = '<>:"/\\|?*'
+    elif os.name == 'posix':  # Unix-like systems
+        invalid_chars = '\0'
+
+    # Check if the file_name contains any invalid characters
+    for char in invalid_chars:
+        if char in file_name:
+            return False
+    return True
```

Comment:
```suggestion
    if os.name == 'nt':  # Windows
        invalid_chars_pattern = r'[<>:"/\\|?*]'
    elif os.name == 'posix':  # Unix-like systems
        invalid_chars_pattern = r'\0'

    # Check if the file_name contains any invalid characters
    return False if re.search(invalid_chars_pattern, file_name) else True
```

This is OK, but it might be more efficient to use a regex instead of a for loop.

Reply:
In this case actually not quite. gpt4o did even a benchmark on this which showed that the loop was twice as fast for this simple use case, compared to a comparable regex approach.
I'd prefer the better performing solution than less readable and less performant (in this case!) regex.